.github/workflows/all-static-checks.yaml

```diff
@@ -34,6 +34,18 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: psf/black@23.10.1
+  check-clang-format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install clang-format
+        run: sudo apt-get install -y clang-format
+      - name: Fetch branch base 
+        if: github.ref != 'refs/heads/main'
```

Comment:
I believe this should be `${{ github.ref !== 'refs/heads/main' }}`, JavaScript style with the braces.

Otherwise this might always evaluate to `true` because it's treated as a string.

Reply:
Also, could we add steps for `main` branch to take the `BASE_COMMIT_SHA` as the one directly preceding the current commit?

We have free compute with the github hosted runners. It'd be nice to have this check occur on every main commit as well upon push, even though we should "technically" get it for free on every PR. Just nice for validation and to see the checkmark.

Reply:
My bad - it should be `!=` - NOT `!==`